internal/levee.go

```diff
@@ -118,3 +123,16 @@ func getArgumentPropagator(c *config.Config, call *ssa.Call) ssa.Node {
 
 	return nil
 }
+
+func isFieldPropagator(c *ssa.Call, fieldPropagators map[types.Object]bool) bool {
```

Comment:
This being here feels awkward to me...

My first thought was that this function should belong to `fieldpropagator/analyzer.go`.  It's unpacking the result of that analysis, so it belongs to that package.

My second thought was that `fieldpropagator.ReturnType` should be a named type instead of an alias, and that this function should be a method on the `ReturnType`.  It's unpacking the result contained in that type, so it belongs to that type.

My third thought was that this is probably fine and we could iterate on it in the future if need be.

I don't know.  Weigh it yourself and I'll trust your judgement on it.

Reply:
Good thinking. I think now is a good time to change it, I'll go with what you suggest in your 2nd thought.